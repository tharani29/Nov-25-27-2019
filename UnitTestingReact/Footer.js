import React, { Component } from 'react';

class Footer extends Component {
  render() {
    return (
      <h5>
        Copyrights reserved, {this.props.year}
      </h5>
    );
  }
}

export default Footer;
