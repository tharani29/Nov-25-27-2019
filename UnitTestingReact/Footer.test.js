import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

describe("Footer", () => {
	it("should render properly with the year prop", () => {
		const component = shallow(<Footer year="2018"/>);
		expect(component).toMatchSnapshot();
	});
	
	it("should render properly with the year prop 2019", () => {
		const component = shallow(<Footer year="2019"/>);
		expect(component).toMatchSnapshot();
	});
});