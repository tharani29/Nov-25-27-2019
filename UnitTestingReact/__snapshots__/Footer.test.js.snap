// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render properly with the year prop 1`] = `
<h5>
  Copyrights reserved, 
  2018
</h5>
`;

exports[`Footer should render properly with the year prop 2019 1`] = `
<h5>
  Copyrights reserved, 
  2019
</h5>
`;
